1_fetch/src/get_nwis_data.R

```diff
@@ -1,22 +1,18 @@
 
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500")){
+download_nwis_data <- function(site_num){
   
   # create the file names that are needed for download_nwis_site_data
   # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
   # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
-  data_out <- data.frame()
-  # loop through files to download 
-  for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
-    # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
-    data_out <- bind_rows(data_out, these_data)
-  }
+  download_file <- file.path(tempdir(), paste0('nwis_', site_num, '_data.csv'))
+  download_nwis_site_data(download_file, parameterCd = '00010')
+  data_out <- read_csv(download_file, col_types = 'ccTdcc')
```

Comment:
I've got two comments here: 

While this write-then-read approach works, I am not sure if it is best practice within a pipeline. The target function both _writes_ and _reads_ a file so the pipeline doesn't have any knowledge of the intermediate file. And maybe that's OK, but it feels like it could have unintended consequences. 

The `download_nwis_data()` function isn't really doing much anymore. Since it is just a wrapper for `download_nwis_site_data()`, you could delete this and directly call `download_nwis_site_data()` from your pipeline code. You could either continue to save as files and pass in filepaths to the function, or edit `download_nwis_site_data()` to accept site numbers and return a data.frame.

Reply:
I removed the wrapper function. My Friday brain totally missed that 😪